lib/src/main/java/io/ably/lib/http/Http.java

```diff
@@ -325,7 +327,17 @@ public void finalize() {
 	 * @throws AblyException
 	 */
 	<T> T ablyHttpExecute(String path, String method, Param[] headers, Param[] params, RequestBody requestBody, ResponseHandler<T> responseHandler) throws AblyException {
-		int retryCountRemaining = Hosts.isRestFallbackSupported(this.host) ? options.httpMaxRetryCount : 0;
+		List<String> customFallbackHosts = null;
+		int retryCountRemaining = 0;
+		if (options.fallbackHosts != null) {
+			if (options.fallbackHosts.length > 0) {
```

Comment:
Why if > 0, shouldn't we set the fallback hosts to an empty array if provided i.e. so that the user can disable fallback if they want?